### ISSUE TITLE
如何在自定义数据集上进行训练或微调？谢谢

### ISSUE THREAD

**Issue by woyaofawz** [liuqk3/PUT#49]

**Comment by woyaofawz** [comment]
tpami2024_p_vqvae_imagenet_res256: train: Epoch 39/100 iter 5470/6400 || generator | adv_loss: 0.0370 | rec_loss: 0.0240 | grad_loss: 0.0367 | content_loss: 0.0266 | style_loss: 0.0802 | loss: 0.2132 | quantize_loss: 0.0087 | used_unmasked_quantize_embed: 436.0000 | used_masked_quantize_embed: 0.0000 | unmasked_num_token: 1024.0000 | masked_num_token: 0.0000 || discriminator | loss: 0.6760 | real_loss: 0.6994 | fake_loss: 0.6526 || generator_lr: 0.0001327449, discriminator_lr: 1.96948e-05 || data_time: 0.0s | fbward_time: 0.4s | iter_time: 0.4s | iter_avg_time: 0.2s | epoch_time: 31m:52s | spend_time: 17h:29m:30s | left_time: 1d:02h:23m:49s
用自己的数据集测试了一下，这个数据是正常的吗？used_masked_quantize_embed: 0.0000 | unmasked_num_token: 1024.0000 | masked_num_token: 0.0000

**Comment by liuqk3** [comment]
Thanks for your interest in our work. I can not remember the loss values since the experiments were carried out about 1.5 years ago. The following are the meanings of some values:

(1) used_masked_quantize_embed: the number of used latent vectors in codebook $e'$ for masked patches. If it equals 0, it means all 1024 image patches are unmasked. Please check the provided masks during the training stages.

(2) unmasked_num_token: the number of unmasked tokens, aka., the number of unmaksed patches.

(3) masked_num_token: the number of masked tokens, aka., the number of maksed patches.

In addition, the training script will save some visualized images during the training stage. You may check the saved images to solve  some puzzles.
